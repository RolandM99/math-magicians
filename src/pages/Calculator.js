import React, { useState } from 'react';
import InputButton from '../components/input';
import calculate from '../logic/calculate';
import '../index.css';

const Calculator = () => {
  const [input, setInput] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleCursor = (clickName) => {
    const click = clickName.target.innerText;
    setInput(calculate(input, click));
  };
  const { total, next, operation } = input;

  const display = () => {
    if (next) {
      return next;
    }
    if (operation) {
      return operation;
    }
    if (total) {
      return total;
    }
    return 0;
  };
  return (
    <div className="main-container">
      <div className="calculator-do">
        <p>Let&apos;s do some mathematics!</p>
      </div>
      <div className="display">{display()}</div>
      <div className="calculator-btn">
        <InputButton value="AC" className="c-btn" click={handleCursor} />
        <InputButton value="+/-" className="c-btn" click={handleCursor} />
        <InputButton value="%" className="c-btn" click={handleCursor} />
        <InputButton value="รท" className="c-btn orange" click={handleCursor} />
        <InputButton value="7" className="c-btn" click={handleCursor} />
        <InputButton value="8" className="c-btn" click={handleCursor} />
        <InputButton value="9" className="c-btn" click={handleCursor} />
        <InputButton value="x" className="c-btn orange" click={handleCursor} />
        <InputButton value="4" className="c-btn" click={handleCursor} />
        <InputButton value="5" className="c-btn" click={handleCursor} />
        <InputButton value="6" className="c-btn" click={handleCursor} />
        <InputButton value="-" className="c-btn orange" click={handleCursor} />
        <InputButton value="1" className="c-btn" click={handleCursor} />
        <InputButton value="2" className="c-btn" click={handleCursor} />
        <InputButton value="3" className="c-btn" click={handleCursor} />
        <InputButton value="+" className="c-btn orange" click={handleCursor} />
        <InputButton value="0" className="c-btn double" click={handleCursor} />
        <InputButton value="." className="c-btn" click={handleCursor} />
        <InputButton value="=" className="c-btn orange" click={handleCursor} />
      </div>
    </div>
  );
};

export default Calculator;
