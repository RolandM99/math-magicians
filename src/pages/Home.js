import React from 'react';

const Home = () => (
  <>
    <div className="home">
      <h1>Welcome to Our Page!</h1>
      <p>
        Mathematics are the result of mysterious powers which no one understands,
        and which the unconscious recognition of beauty must play an important part.
        Out of an infinity of designs a mathematician chooses one pattern for
        beautyâ€™s sake and pulls it down to earth.
      </p>
      <p>
        Mathematics as an expression of the human mind reflects the active will,
        the contemplative reason, and the desire for aesthetic perfection.
        Its basic elements are logic and intuition, analysis and construction,
        generality and individuality.
      </p>
      <p>
        It is impossible to be a mathematician without being a poet in soul&#128525;
      </p>
      <br />
      <p>Let&apos;s have fun with Math-Magician&#128513;&#128517;</p>
    </div>
  </>
);

export default Home;
