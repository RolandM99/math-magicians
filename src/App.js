import React from 'react';
import {
  BrowserRouter,
  Routes,
  Route,
} from 'react-router-dom';
import Calculator from './pages/Calculator';
import Home from './pages/Home';
import NavBar from './components/Navbar';
import Quotes from './pages/Quotes';

function App() {
  return (
    <BrowserRouter>
      <NavBar />
      <Routes>
        <Route
          path="/calculator"
          element={<Calculator />}
        />
        <Route
          path="/quotes"
          element={<Quotes />}
        />
        <Route
          path="/"
          element={<Home />}
        />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
