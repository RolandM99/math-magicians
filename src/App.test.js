import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Calculator from './pages/Calculator';
import Quote from './pages/Quotes';
import Home from './pages/Home';

test('Checking of the sentence Math Magician', () => {
  render(<App />);
  const linkElement = screen.getByText(/Math Magician/i);
  expect(linkElement).toMatchSnapshot();
});
test('change screen when we click in the navbar on Calculator', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('change screen when we click in the navbar on Quotes', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('change screen when we click in the navbar on Home', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
