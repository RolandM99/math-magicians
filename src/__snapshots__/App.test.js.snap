// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking of the sentence Math Magician 1`] = `
<h1>
  Math Magician
</h1>
`;

exports[`change screen when we click in the navbar on Calculator 1`] = `
<div
  className="main-container"
>
  <div
    className="calculator-do"
  >
    <p>
      Let's do some mathematics!
    </p>
  </div>
  <div
    className="display"
  >
    0
  </div>
  <div
    className="calculator-btn"
  >
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="c-btn orange"
      onClick={[Function]}
      type="button"
    >
      √∑
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="c-btn orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="c-btn orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="c-btn orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="c-btn double"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="c-btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="c-btn orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`change screen when we click in the navbar on Home 1`] = `
<div
  className="home"
>
  <h1>
    Welcome to Our Page!
  </h1>
  <p>
    Mathematics are the result of mysterious powers which no one understands, and which the unconscious recognition of beauty must play an important part. Out of an infinity of designs a mathematician chooses one pattern for beauty‚Äôs sake and pulls it down to earth.
  </p>
  <p>
    Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
  </p>
  <p>
    It is impossible to be a mathematician without being a poet in soulüòç
  </p>
  <br />
  <p>
    Let's have fun with Math-MagicianüòÅüòÖ
  </p>
</div>
`;

exports[`change screen when we click in the navbar on Quotes 1`] = `
<div
  className="quotes"
>
  <blockquote>
    Mathematics has beauty and romance. It‚Äôs not a boring place to be, the mathematical world. It‚Äôs an extraordinary place; it‚Äôs worth spending time there.
  </blockquote>
  <strong>
    ~ Marcus du Sautoy
  </strong>
</div>
`;

exports[`renders learn react link 1`] = `
<h1>
  Math Magician
</h1>
`;
